### ISSUE TITLE
Rakudo-2022.06 with JVM backend build failure

### ISSUE THREAD

**Issue by fabrice-ducos** [rakudo/rakudo#4952]
## The Problem
I am trying to build `rakudo-2022.06` with the JVM backend.

## Expected Behavior
The build completes.

## Actual Behavior
The build fails with the following error.

```
+++ Compiling	rakudo.jar
+++ Generating	gen/jvm/BOOTSTRAP/v6c.nqp
+++ Compiling	blib/Perl6/BOOTSTRAP/v6c.jar
fatal: pas dans un répertoire git
fatal: pas dans un répertoire git
+++ Compiling	blib/CORE.c.setting.jar
fatal: pas dans un répertoire git
fatal: pas dans un répertoire git
The following step can take a long time, please be patient.
Stage start      :   0.000
Stage parse      : 179.591
Stage syntaxcheck:   0.000
Stage ast        :   0.000
Stage optimize   :  20.098
Stage jast       :  40.676
Stage classfile  :  14.729
Stage jar        :   0.000
+++ Generating	gen/jvm/BOOTSTRAP/v6d.nqp
+++ Compiling	blib/Perl6/BOOTSTRAP/v6d.jar
+++ Compiling	blib/CORE.d.setting.jar
fatal: pas dans un répertoire git
fatal: pas dans un répertoire git
The following step can take a long time, please be patient.
Stage start      :   0.000
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  Internal Error (metaspaceArena.cpp:93), pid=20966, tid=9731
#  guarantee(requested_word_size <= chunklevel::MAX_CHUNK_WORD_SIZE) failed: Requested size too large (557969) - max allowed size per allocation is 524288.
#
# JRE version: OpenJDK Runtime Environment Homebrew (17.0.2) (build 17.0.2+0)
# Java VM: OpenJDK 64-Bit Server VM Homebrew (17.0.2+0, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, bsd-amd64)
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/ducos/Developer/perl6/rakudo-2022.06/hs_err_pid20966.log
#
# If you would like to submit a bug report, please visit:
#   https://github.com/Homebrew/homebrew-core/issues
#
```

## Steps to Reproduce
`perl Configure.pl --gen-nqp --backends=jvm && make && make test`
(it never reaches "make test", it fails at the "make" step)

## Environment
* Operating system: MacOSX Monterey (12.4) and OpenJDK 64-Bit Server VM Homebrew (17.0.2+)
* Compiler version (`perl6 -v` or `raku -v`):

**Comment by Kaiepi** [comment]
I kinda need to be able to build the JVM backend to test new atomic ops?

**Comment by Kaiepi** [comment]
The JVM is go on Alpine Linux in https://github.com/rakudo/rakudo/pull/5000. Other platforms may still have trouble at this point.

**Comment by fabrice-ducos** [comment]
Some more results about this issue. I am still testing on MacOS Monterey (12.4).
I tried to build the current version available in the master branch of the git repo (at the time of writing).
With the **JDK 17** (openjdk64-17.0.3 to be accurate), *it fails*, with the same error as in the initial post.

With the **JDK 11** (openjdk64-11.0.6), *it builds*. There are some errors in the tests though (see below).

It would be interesting to test on other systems (Linux, Windows...) with the JDK 17, if it was not done already.

Here is the interpreter I obtain with the JDK 11:

```
$ ./rakudo
Welcome to Rakudo™ v2022.06.
Implementing the Raku® Programming Language v6.d.
Built on JVM.

You may want to `zef install Readline`, `zef install Linenoise`, or `zef install Terminal::LineEditor` or use rlwrap for a line editor

To exit type 'exit' or '^D'
[0] > print "Hello Rakudo\n";
Hello Rakudo
```

However, the test suite eventually **FAILS** even if the interpreter seems to be built successfully with the JDK 11:

```
$ perl Configure.pl --gen-nqp --backends=jvm && make && make test
===ATTENTION===
  No --prefix supplied, building and installing to /Users/ducos/Developer/perl6/rakudo/install
Using /Users/ducos/Developer/perl6/rakudo/install/bin/nqp-j (version 2022.06-1-ge3c439991 / OpenJDK 11.0.6).
Cleaning up ...

You can now use 'make' to build Rakudo.
After that, 'make test' will run some tests and
'make install' will install Rakudo.
+++ Checking for jvm NQP version
+++ Expanding	gen/jvm/main-version.nqp
+++ Generating	gen/jvm/rakudo.nqp
+++ Generating	gen/jvm/Grammar.nqp
+++ Generating	gen/jvm/World.nqp
+++ Generating	gen/jvm/ModuleLoader.nqp
+++ Compiling	blib/Perl6/ModuleLoader.jar
+++ Generating	gen/jvm/Ops.nqp
+++ Compiling	blib/Perl6/Ops.jar
+++ Generating	gen/jvm/Pod.nqp
+++ Compiling	blib/Perl6/Pod.jar
+++ Compiling	blib/Perl6/World.jar
+++ Generating	gen/jvm/Actions.nqp
+++ Compiling	blib/Perl6/Actions.jar
+++ Compiling	blib/Perl6/Grammar.jar
+++ Generating	gen/jvm/Compiler.nqp
+++ Generating	gen/jvm/Optimizer.nqp
+++ Generating	gen/jvm/Metamodel.nqp
+++ Compiling	blib/Perl6/Metamodel.jar
+++ Compiling	blib/Perl6/Optimizer.jar
+++ Compiling	blib/Perl6/Compiler.jar
+++ Generating	gen/jvm/SysConfig.nqp
+++ Compiling	blib/Perl6/SysConfig.jar
+++ Generating	rakudo-runtime.jar
src/vm/jvm/runtime/org/raku/rakudo/RakudoContainerSpec.java:4: warning: Unsafe is internal proprietary API and may be removed in a future release
import sun.misc.Unsafe;
               ^
src/vm/jvm/runtime/org/raku/rakudo/RakudoContainerSpec.java:94: warning: Unsafe is internal proprietary API and may be removed in a future release
    private Unsafe unsafe;
            ^
src/vm/jvm/runtime/org/raku/rakudo/RakudoContainerSpec.java:101: warning: Unsafe is internal proprietary API and may be removed in a future release
                Field unsafeField = Unsafe.class.getDeclaredField("theUnsafe");
                                    ^
src/vm/jvm/runtime/org/raku/rakudo/RakudoContainerSpec.java:103: warning: Unsafe is internal proprietary API and may be removed in a future release
                unsafe = (Unsafe)unsafeField.get(null);
                          ^
Note: src/vm/jvm/runtime/org/raku/rakudo/RakudoJavaInterop.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: src/vm/jvm/runtime/org/raku/rakudo/RakudoJavaInterop.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
4 warnings
+++ Compiling	rakudo.jar
+++ Generating	gen/jvm/BOOTSTRAP/v6c.nqp
+++ Compiling	blib/Perl6/BOOTSTRAP/v6c.jar
+++ Compiling	blib/CORE.c.setting.jar
The following step can take a long time, please be patient.
Stage start      :   0.000
Stage parse      : 266.608
Stage syntaxcheck:   0.000
Stage ast        :   0.001
Stage optimize   :  24.555
Stage jast       :  48.723
Stage classfile  :  14.466
Stage jar        :   0.000
+++ Generating	gen/jvm/BOOTSTRAP/v6d.nqp
+++ Compiling	blib/Perl6/BOOTSTRAP/v6d.jar
+++ Compiling	blib/CORE.d.setting.jar
The following step can take a long time, please be patient.
Stage start      :   0.001
Stage parse      :   5.685
Stage syntaxcheck:   0.000
Stage ast        :   0.000
Stage optimize   :   0.114
Stage jast       :   0.441
Stage classfile  :   0.122
Stage jar        :   0.000
+++ Generating	gen/jvm/BOOTSTRAP/v6e.nqp
+++ Compiling	blib/Perl6/BOOTSTRAP/v6e.jar
+++ Compiling	blib/CORE.e.setting.jar
The following step can take a long time, please be patient.
Stage start      :   0.001
Stage parse      :  11.850
Stage syntaxcheck:   0.000
Stage ast        :   0.000
Stage optimize   :   0.552
Stage jast       :   2.102
Stage classfile  :   0.363
Stage jar        :   0.000
+++ Setting up	rakudo-j
+++ Generating	gen/jvm/rakudo-debug.nqp
+++ Compiling	rakudo-debug.jar
+++ Setting up	rakudo-debug-j
+++ Setting up	eval-client.pl
+++ Setting up JVM runner
+++ Checking for jvm NQP version
'/usr/local/Cellar/perl/5.34.0/bin/perl' -I'/Users/ducos/Developer/perl6/rakudo/tools/lib' -I'/Users/ducos/Developer/perl6/rakudo/3rdparty/nqp-configure/lib' t/harness5 --jvm t/01-sanity t/02-rakudo t/04-nativecall t/05-messages t/06-telemetry t/07-pod-to-text t/08-performance t/10-qast t/03-jvm
t/01-sanity/01-literals.t ................... ok                        
t/01-sanity/02-op-math.t .................... ok                        
t/01-sanity/05-var.t ........................ ok                        
t/01-sanity/04-op-cmp.t ..................... ok                        
t/01-sanity/03-op-logic.t ................... ok                        
t/01-sanity/06-op-inplace.t ................. ok                        
t/01-sanity/07-op-string.t .................. ok                        
t/01-sanity/11-tap.t ........................ ok                        
t/01-sanity/08-var-array.t .................. ok                        
t/01-sanity/09-end-blocks.t ................. ok                        
t/01-sanity/10-regex.t ...................... ok                        
t/01-sanity/12-counter.t .................... ok                        
t/01-sanity/13-equal.t ...................... ok                        
t/01-sanity/16-eqv.t ........................ ok                        
t/01-sanity/15-sub.t ........................ ok                        
t/01-sanity/17-isa.t ........................ ok                        
t/01-sanity/18-simple-multisubs.t ........... ok                        
t/01-sanity/14-if.t ......................... ok                        
t/01-sanity/19-say.t ........................ ok                        
t/01-sanity/20-defined.t .................... ok                        
t/01-sanity/21-try.t ........................ ok                        
t/01-sanity/53-transpose.t .................. ok                        
t/01-sanity/98-test-deprecated.t ............ ok                        
t/02-rakudo/01-is_approx.t .................. ok                        
t/01-sanity/56-use-isms.t ................... ok                        
t/01-sanity/99-test-basic.t ................. ok                        
t/01-sanity/55-use-trace.t .................. ok                        
t/02-rakudo/03-corekeys-6c.t ................ ok                        
t/02-rakudo/03-corekeys-6d.t ................ ok                        
t/02-rakudo/03-corekeys.t ................... ok                        
t/02-rakudo/03-corekeys-6e.t ................ ok                        
t/02-rakudo/04-settingkeys-6d.t ............. ok                        
t/02-rakudo/04-diag.t ....................... ok                        
t/02-rakudo/05-range-in-range.t ............. ok                        
t/02-rakudo/04-settingkeys-6c.t ............. ok                        
t/02-rakudo/06-is.t ......................... ok                        
t/02-rakudo/04-settingkeys-6e.t ............. ok                        
t/02-rakudo/03-cmp-ok.t ..................... ok                        
t/02-rakudo/07-implementation-detail-6.d.t .. ok                        
t/02-rakudo/07-implementation-detail-6.c.t .. ok                        
t/02-rakudo/07-implementation-detail-6.e.t .. ok                        
t/02-rakudo/08-inline-native-arith.t ........ ok                        
t/02-rakudo/09-thread-id-after-await.t ...... ok                        
t/02-rakudo/10-nqp-ops.t .................... ok                        
t/02-rakudo/08-slangs.t ..................... ok                        
t/02-rakudo/13-exceptions.t ................. ok                        
t/02-rakudo/11-deprecated.t ................. ok                        
t/02-rakudo/12-proto-arity-count.t .......... skipped: some subs are not defined on JVM backend
t/02-rakudo/16-begin-time-eval.t ............ skipped: java.lang.ArrayIndexOutOfBoundsException: Index 7 out of bounds for length 7
t/02-rakudo/17-cli.t ........................ ok                        
t/02-rakudo/19-mkdir.t ...................... ok                        
t/02-rakudo/08-repeat.t ..................... ok                        
t/02-rakudo/18-pseudostash.t ................ ok                        
t/02-rakudo/dump.t .......................... ok                        
t/02-rakudo/dd.t ............................ ok                        
t/02-rakudo/v6.d-tests/01-deprecations.t .... ok                        
t/03-jvm/01-interop.t ....................... ok                        
t/02-rakudo/99-misc.t ....................... ok                        
t/02-rakudo/14-revisions.t .................. ok                        
t/04-nativecall/02-simple-args.t ............ ok                        
t/04-nativecall/00-misc.t ................... skipped: NullPointerException in sub NCstrlen
t/04-nativecall/01-argless.t ................ ok                        
t/04-nativecall/05-arrays.t ................. skipped: NullPointerException in sub ReturnADoubleArray
t/04-nativecall/04-pointers.t ............... ok                        
t/02-rakudo/15-gh_1202.t .................... Dubious, test returned 1 (wstat 256, 0x100)
Failed 2/2 subtests 
t/04-nativecall/03-simple-returns.t ......... ok                        
t/02-rakudo/reproducible-builds.t ........... ok                        
t/04-nativecall/07-writebarrier.t ........... skipped: NullPointerException in sub _deref
t/04-nativecall/06-struct.t ................. ok                        
t/04-nativecall/08-callbacks.t .............. ok                        
t/04-nativecall/10-cglobals.t ............... ok                        
t/04-nativecall/09-nativecast.t ............. ok                        
t/04-nativecall/11-cpp.t .................... ok                        
t/04-nativecall/12-sizeof.t ................. ok                        
t/04-nativecall/13-cpp-mangling.t ........... ok                        
t/04-nativecall/13-union.t .................. ok                        
t/04-nativecall/14-rw-attrs.t ............... ok                        
t/04-nativecall/16-rt125408.t ............... ok                        
t/04-nativecall/16-rt125729.t ............... ok                        
t/04-nativecall/17-libnames.t ............... ok                        
t/04-nativecall/15-rw-args.t ................ Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 12/24 subtests 
t/04-nativecall/18-routine-sig-sanity.t ..... ok                        
t/04-nativecall/19-function-pointers.t ...... ok                        
t/04-nativecall/21-callback-other-thread.t .. ok                        
t/04-nativecall/22-method.t ................. skipped: NullPointerException in sub ReturnAStruct
t/04-nativecall/20-concurrent.t ............. ok                        
t/04-nativecall/23-incomplete-types.t ....... ok                        
t/05-messages/11-overflow.t ................. skipped: These tests do not throw on JVM backend
t/05-messages/moar/01-errors.t .............. skipped: These tests are only for MoarVM backend
t/05-messages/v6.d-tests/01-errors.t ........ ok                        
t/05-messages/01-errors.t ................... ok                        
t/05-messages/10-warnings.t ................. ok                        
t/06-telemetry/02-usage.t ................... ok                        
t/06-telemetry/03-thread.t .................. ok                        
t/06-telemetry/01-basic.t ................... ok                        
t/05-messages/03-errors.t ................... ok                        
t/06-telemetry/04-threadpool.t .............. ok                        
t/08-performance/01-iterators.t ............. ok                        
t/08-performance/05-processkeys.t ........... ok                        
t/07-pod-to-text/01-whitespace.t ............ ok                        
t/07-pod-to-text/02-input-output.t .......... ok                        
t/08-performance/99-misc.t .................. ok                        
t/10-qast/00-misc.t ......................... ok                        
t/08-performance/02-qast-rewrites.t ......... ok                        
t/05-messages/02-errors.t ................... ok                        
t/02-rakudo/repl.t .......................... ok     

Test Summary Report
-------------------
t/02-rakudo/18-pseudostash.t              (Wstat: 0 Tests: 3 Failed: 0)
  TODO passed:   1-3
t/02-rakudo/15-gh_1202.t                  (Wstat: 256 Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
  Parse errors: Bad plan.  You planned 2 tests but ran 1.
t/04-nativecall/15-rw-args.t              (Wstat: 65280 Tests: 12 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 24 tests but ran 12.
Files=106, Tests=1452, 1908 wallclock secs ( 1.05 usr  0.42 sys + 14323.77 cusr 426.53 csys = 14751.77 CPU)
Result: FAIL
make: *** [j-coretest5] Error 1
```

**Comment by Kaiepi** [comment]
Same TODOs pass, same tests fail on the `counter` branch at the moment.

**Comment by Kaiepi** [comment]
I set `ulimit -c unlimited` to allow a core dump to be generated with OpenJDK 17. It's trying to read too large a file:

```java
LAPTOP-7I9RCT0C:~/Documents/raku/rakudo% jdb /mnt/wslg/dumps/core.java
Initializing jdb ...
> run
run /mnt/wslg/dumps/core.java
Set uncaught java.lang.Throwable
Set deferred uncaught java.lang.Throwable
>
VM Started:
Exception occurred: java.lang.OutOfMemoryError (uncaught)"thread=main", com.sun.tools.javac.launcher.Main.readFile(), line=235 bci=236

main[1] where
  [1] com.sun.tools.javac.launcher.Main.readFile (Main.java:235)
  [2] com.sun.tools.javac.launcher.Main.compile (Main.java:371)
  [3] com.sun.tools.javac.launcher.Main.run (Main.java:189)
  [4] com.sun.tools.javac.launcher.Main.main (Main.java:132)
```

Unsure exactly which file exactly, but I suspect it'd be the `CORE.c.setting.jar` dependency. We'd need to serialize more efficiently or find a setup that allows for a larger memory limit otherwise should that be the case.

**Comment by usev6** [comment]
Just for the record: The mentioned issue for OpenJDK has been tagged as "affects version 17" and "solved for version 21".

The issue also has two links to commits that seem to backport the fix to jdk-17and jdk-22. 

So it's probably worth a try to see if the error is still there with the latest OpenJDK 17 (e.g. 17.0.8 from Adoptium: https://github.com/adoptium/temurin17-binaries).
(Disclaimer: I can't promise that I'll have time to try that.)

**Comment by MasterDuke17** [comment]
It's still there with OpenJDK 20 (from the extra/jdk-openjdk 20.0.2.u9-3 package on Arch linux).
```
Stage start      :   0.000
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  Internal Error (metaspaceArena.cpp:86), pid=3161991, tid=3161992
#  guarantee(requested_word_size <= chunklevel::MAX_CHUNK_WORD_SIZE) failed: Requested size too large (616370) - max allowed size per allocation is 524288.
#
# JRE version: OpenJDK Runtime Environment (20.0.2+9) (build 20.0.2+9)
# Java VM: OpenJDK 64-Bit Server VM (20.0.2+9, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, parallel gc, linux-amd64)
```

**Comment by usev6** [comment]
I just tried on Debian 11 with ```jdk-17.0.8.1+1``` taken from https://adoptium.net/de/temurin/releases/ and didn't see an error building ```rakudo-j```.

(There is a deprecation warning caused by ```-noverify``` being used, but that's a different story.)
```
$ ./rakudo-j --version
OpenJDK 64-Bit Server VM warning: Options -Xverify:none and -noverify were deprecated in JDK 13 and will likely be removed in a future release.
Welcome to Rakudo™ v2023.08-185-gca05e9fbe.
Implementing the Raku® Programming Language v6.d.
Built on JVM.
$ ./rakudo-j -e 'say $*VM.properties<java.runtime.version>'
OpenJDK 64-Bit Server VM warning: Options -Xverify:none and -noverify were deprecated in JDK 13 and will likely be removed in a future release.
17.0.8.1+1
```

**Comment by tstuefe** [comment]
> It's still there with OpenJDK 20 (from the extra/jdk-openjdk 20.0.2.u9-3 package on Arch Linux).
> 

I fixed it for 21 onwards, and downported it to JDK 17. JDK 20 is still broken, and since it will phased out soon, I don't plan on fixing it.

**Comment by fabrice-ducos** [comment]
Dear Rakudo Team,

Thank you very much for this continuous effort.
I tested again with OpenJDK 17 and it now works for me.

There are still some errors during tests (see below) but the interpreter is properly built.

As the OP of this thread, I consider this issue fixed and will close it.

Thanks again and best regards.

-----------------------

`% perl Configure.pl --gen-nqp --backends=jvm && make && make test`

```
===ATTENTION===
  No --prefix supplied, building and installing to /Users/ducos/Downloads/rakudo-2023.09/install
No pre-existing installed file found at /Users/ducos/Downloads/rakudo-2023.09/install/bin/nqp-j
Cloning from https://github.com/Raku/nqp.git
Cloning into 'nqp'...
remote: Enumerating objects: 97267, done.
remote: Counting objects: 100% (3892/3892), done.
remote: Compressing objects: 100% (1263/1263), done.
remote: Total 97267 (delta 2422), reused 3650 (delta 2301), pack-reused 93375
Receiving objects: 100% (97267/97267), 154.44 MiB | 9.89 MiB/s, done.
Resolving deltas: 100% (63052/63052), done.
Note: switching to '2023.09'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 61b2ed682 [release] Bump VERSION to 2023.09
Building NQP ...
perl Configure.pl --prefix=/Users/ducos/Downloads/rakudo-2023.09/install --make-install --git-protocol=https --no-relocatable --no-ignore-errors --silent-build --no-force-rebuild --backends=jvm
Updating submodules .................................... OK
got: openjdk version "17.0.8" 2023-07-18
Using openjdk version "17.0.8" 2023-07-18

Cleaning up ...
+++ Preparing JVM build directories
++++++ Building JVM backend
+++ Preparing Java runtime
src/vm/jvm/runtime/org/raku/nqp/runtime/Ops.java:146: warning: Unsafe is internal proprietary API and may be removed in a future release
import sun.misc.Unsafe;
               ^
src/vm/jvm/runtime/org/raku/nqp/sixmodel/reprs/P6OpaqueBaseInstance.java:10: warning: Unsafe is internal proprietary API and may be removed in a future release
import sun.misc.Unsafe;
               ^
src/vm/jvm/runtime/org/raku/nqp/runtime/Ops.java:162: warning: Unsafe is internal proprietary API and may be removed in a future release
            Field theUnsafe = Unsafe.class.getDeclaredField("theUnsafe");
                              ^
src/vm/jvm/runtime/org/raku/nqp/runtime/Ops.java:164: warning: Unsafe is internal proprietary API and may be removed in a future release
            Unsafe u = (Unsafe)theUnsafe.get(null);
            ^
src/vm/jvm/runtime/org/raku/nqp/runtime/Ops.java:164: warning: Unsafe is internal proprietary API and may be removed in a future release
            Unsafe u = (Unsafe)theUnsafe.get(null);
                        ^
src/vm/jvm/runtime/org/raku/nqp/sixmodel/reprs/P6OpaqueBaseInstance.java:119: warning: Unsafe is internal proprietary API and may be removed in a future release
    private Unsafe unsafe;
            ^
src/vm/jvm/runtime/org/raku/nqp/sixmodel/reprs/P6OpaqueBaseInstance.java:126: warning: Unsafe is internal proprietary API and may be removed in a future release
                Field unsafeField = Unsafe.class.getDeclaredField("theUnsafe");
                                    ^
src/vm/jvm/runtime/org/raku/nqp/sixmodel/reprs/P6OpaqueBaseInstance.java:128: warning: Unsafe is internal proprietary API and may be removed in a future release
                unsafe = (Unsafe)unsafeField.get(null);
                          ^
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
8 warnings
+++ Generating	gen/jvm/stage1/nqpmo.nqp
+++ Compiling	gen/jvm/stage1/nqpmo.jar
+++ Compiling	gen/jvm/stage1/ModuleLoader.jar
+++ Generating	gen/jvm/stage1/NQPCORE.setting
+++ Compiling	gen/jvm/stage1/NQPCORE.setting.jar
+++ Generating	gen/jvm/stage1/QASTNode.nqp
+++ Compiling	gen/jvm/stage1/QASTNode.jar
+++ Generating	gen/jvm/stage1/QRegex.nqp
+++ Compiling	gen/jvm/stage1/QRegex.jar
+++ Generating 	stage 1 nqp-config.nqp
+++ Generating	gen/jvm/stage1/NQPHLL.nqp
+++ Generating	gen/jvm/stage1/JASTNodes.nqp
+++ Compiling	gen/jvm/stage1/JASTNodes.jar
+++ Compiling	gen/jvm/stage1/NQPHLL.jar
+++ Generating	gen/jvm/stage1/QAST.nqp
+++ Compiling	gen/jvm/stage1/QAST.jar
+++ Generating	gen/jvm/stage1/NQPP6QRegex.nqp
+++ Compiling	gen/jvm/stage1/NQPP6QRegex.jar
+++ Creating	stage 1 NQP
+++ Generating	gen/jvm/stage2/nqpmo.nqp
+++ Compiling	gen/jvm/stage2/nqpmo.jar
+++ Compiling	gen/jvm/stage2/ModuleLoader.jar
+++ Generating	gen/jvm/stage2/NQPCORE.setting
+++ Compiling	gen/jvm/stage2/NQPCORE.setting.jar
+++ Generating	gen/jvm/stage2/QASTNode.nqp
+++ Compiling	gen/jvm/stage2/QASTNode.jar
+++ Generating	gen/jvm/stage2/QRegex.nqp
+++ Compiling	gen/jvm/stage2/QRegex.jar
+++ Generating 	stage 2 nqp-config.nqp
+++ Generating	gen/jvm/stage2/NQPHLL.nqp
+++ Generating	gen/jvm/stage2/JASTNodes.nqp
+++ Compiling	gen/jvm/stage2/JASTNodes.jar
+++ Compiling	gen/jvm/stage2/NQPHLL.jar
+++ Generating	gen/jvm/stage2/QAST.nqp
+++ Compiling	gen/jvm/stage2/QAST.jar
+++ Generating	gen/jvm/stage2/NQPP6QRegex.nqp
+++ Compiling	gen/jvm/stage2/NQPP6QRegex.jar
+++ Creating	stage 2 NQP
+++ Installing JVM stage 2 files locally
+++ Setting up	nqp-j
got: openjdk version "17.0.8" 2023-07-18
Using openjdk version "17.0.8" 2023-07-18

+++ Compiling	NQPP5QRegex.jar
+++ Setting up default runner
+++ Preparing JVM build directories
++++++ Building JVM backend
+++ Installing JVM stage 2 files locally
+++ Setting up	nqp-j
got: openjdk version "17.0.8" 2023-07-18
Using openjdk version "17.0.8" 2023-07-18

+++ Creating JVM installation directories
+++ Installing into /Users/ducos/Downloads/rakudo-2023.09/install
+++ Installing JVM specifics
got: openjdk version "17.0.8" 2023-07-18
Using openjdk version "17.0.8" 2023-07-18

+++ Setting up default runner
+++ Installing default runner

NQP has been built and installed.
Using /Users/ducos/Downloads/rakudo-2023.09/install/bin/nqp-j (version 2023.09 / OpenJDK 17.0.8).
Cleaning up ...

You can now use 'make' to build Rakudo.
After that, 'make test' will run some tests and
'make install' will install Rakudo.
+++ Checking for jvm NQP version
+++ Expanding	gen/jvm/main-version.nqp
+++ Generating	gen/jvm/rakudo.nqp
+++ Generating	gen/jvm/Grammar.nqp
+++ Generating	gen/jvm/World.nqp
+++ Generating	gen/jvm/ModuleLoader.nqp
+++ Compiling	blib/Perl6/ModuleLoader.jar
+++ Generating	gen/jvm/Ops.nqp
+++ Compiling	blib/Perl6/Ops.jar
+++ Generating	gen/jvm/Pod.nqp
+++ Compiling	blib/Perl6/Pod.jar
+++ Compiling	blib/Perl6/World.jar
+++ Generating	gen/jvm/Actions.nqp
+++ Compiling	blib/Perl6/Actions.jar
+++ Compiling	blib/Perl6/Grammar.jar
+++ Generating	gen/jvm/RakuGrammar.nqp
+++ Generating	gen/jvm/RakuActions.nqp
+++ Compiling	blib/Raku/Actions.jar
+++ Compiling	blib/Raku/Grammar.jar
+++ Generating	gen/jvm/Compiler.nqp
+++ Generating	gen/jvm/Optimizer.nqp
+++ Generating	gen/jvm/Metamodel.nqp
+++ Compiling	blib/Perl6/Metamodel.jar
+++ Compiling	blib/Perl6/Optimizer.jar
+++ Compiling	blib/Perl6/Compiler.jar
+++ Generating	gen/jvm/SysConfig.nqp
+++ Compiling	blib/Perl6/SysConfig.jar
+++ Generating	rakudo-runtime.jar
src/vm/jvm/runtime/org/raku/rakudo/RakudoContainerSpec.java:4: warning: Unsafe is internal proprietary API and may be removed in a future release
import sun.misc.Unsafe;
               ^
src/vm/jvm/runtime/org/raku/rakudo/RakudoContainerSpec.java:94: warning: Unsafe is internal proprietary API and may be removed in a future release
    private Unsafe unsafe;
            ^
src/vm/jvm/runtime/org/raku/rakudo/RakudoContainerSpec.java:101: warning: Unsafe is internal proprietary API and may be removed in a future release
                Field unsafeField = Unsafe.class.getDeclaredField("theUnsafe");
                                    ^
src/vm/jvm/runtime/org/raku/rakudo/RakudoContainerSpec.java:103: warning: Unsafe is internal proprietary API and may be removed in a future release
                unsafe = (Unsafe)unsafeField.get(null);
                          ^
Note: src/vm/jvm/runtime/org/raku/rakudo/RakudoJavaInterop.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: src/vm/jvm/runtime/org/raku/rakudo/RakudoJavaInterop.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
4 warnings
+++ Compiling	rakudo.jar
+++ Generating	gen/jvm/ast.nqp
+++ Generating	gen/jvm/BOOTSTRAP/v6c.nqp
+++ Compiling	blib/Perl6/BOOTSTRAP/v6c.jar
+++ Compiling	blib/CORE.c.setting.jar
The following step can take a long time, please be patient.
Stage start      :   0.000
Stage parse      : 1117.168
Stage syntaxcheck:   0.000
Stage ast        :   0.000
Stage optimize   :  13.666
Stage jast       :  27.053
Stage classfile  :   6.707
Stage jar        :   0.000
+++ Generating	gen/jvm/BOOTSTRAP/v6d.nqp
+++ Compiling	blib/Perl6/BOOTSTRAP/v6d.jar
+++ Compiling	blib/CORE.d.setting.jar
The following step can take a long time, please be patient.
Stage start      :   0.001
Stage parse      :   4.827
Stage syntaxcheck:   0.000
Stage ast        :   0.000
Stage optimize   :   0.076
Stage jast       :   0.269
Stage classfile  :   0.072
Stage jar        :   0.000
+++ Generating	gen/jvm/BOOTSTRAP/v6e.nqp
+++ Compiling	blib/Perl6/BOOTSTRAP/v6e.jar
+++ Compiling	blib/CORE.e.setting.jar
The following step can take a long time, please be patient.
Stage start      :   0.000
Stage parse      :  11.700
Stage syntaxcheck:   0.000
Stage ast        :   0.000
Stage optimize   :   0.651
Stage jast       :   2.444
Stage classfile  :   0.367
Stage jar        :   0.000
+++ Setting up	rakudo-j
+++ Generating	gen/jvm/rakudo-debug.nqp
+++ Compiling	rakudo-debug.jar
+++ Setting up	rakudo-debug-j
+++ Setting up	eval-client.pl
+++ Setting up JVM runner
'perl' -I'/Users/ducos/Downloads/rakudo-2023.09/tools/lib' -I'/Users/ducos/Downloads/rakudo-2023.09/3rdparty/nqp-configure/lib' t/harness5 --jvm t/01-sanity t/02-rakudo t/04-nativecall t/05-messages t/06-telemetry t/07-pod-to-text t/08-performance t/10-qast t/03-jvm
t/01-sanity/01-literals.t ................... ok
t/01-sanity/03-op-logic.t ................... ok
t/01-sanity/02-op-math.t .................... ok
t/01-sanity/05-var.t ........................ ok
t/01-sanity/04-op-cmp.t ..................... ok
t/01-sanity/06-op-inplace.t ................. ok
t/01-sanity/09-end-blocks.t ................. ok
t/01-sanity/08-var-array.t .................. ok
t/01-sanity/11-tap.t ........................ ok
t/01-sanity/12-counter.t .................... ok
t/01-sanity/07-op-string.t .................. ok
t/01-sanity/10-regex.t ...................... ok
t/01-sanity/13-equal.t ...................... ok
t/01-sanity/15-sub.t ........................ ok
t/01-sanity/16-eqv.t ........................ ok
t/01-sanity/18-simple-multisubs.t ........... ok
t/01-sanity/17-isa.t ........................ ok
t/01-sanity/14-if.t ......................... ok
t/01-sanity/19-say.t ........................ ok
t/01-sanity/20-defined.t .................... ok
t/01-sanity/21-try.t ........................ ok
t/01-sanity/53-transpose.t .................. ok
t/01-sanity/56-use-isms.t ................... ok
t/01-sanity/99-test-basic.t ................. ok
t/01-sanity/55-use-trace.t .................. Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/3 subtests
t/02-rakudo/03-corekeys-6e.t ................ ok
t/02-rakudo/03-corekeys-6d.t ................ ok
t/02-rakudo/03-corekeys-6c.t ................ ok
t/02-rakudo/03-corekeys.t ................... ok
t/02-rakudo/04-settingkeys-6d.t ............. ok
t/02-rakudo/05-range-in-range.t ............. ok
t/02-rakudo/04-diag.t ....................... ok
t/02-rakudo/04-settingkeys-6c.t ............. ok
t/02-rakudo/06-is.t ......................... ok
t/02-rakudo/07-implementation-detail-6.c.t .. ok
t/02-rakudo/03-cmp-ok.t ..................... ok
t/02-rakudo/07-implementation-detail-6.d.t .. ok
t/02-rakudo/04-settingkeys-6e.t ............. ok
t/02-rakudo/07-implementation-detail-6.e.t .. ok
t/02-rakudo/08-inline-native-arith.t ........ ok
t/02-rakudo/09-thread-id-after-await.t ...... ok
t/02-rakudo/10-nqp-ops.t .................... Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/2 subtests
t/02-rakudo/08-slangs.t ..................... Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests
t/02-rakudo/11-deprecated.t ................. Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/2 subtests
t/02-rakudo/13-exceptions.t ................. ok
t/02-rakudo/12-proto-arity-count.t .......... skipped: some subs are not defined on JVM backend
t/02-rakudo/16-begin-time-eval.t ............ skipped: java.lang.ArrayIndexOutOfBoundsException: Index 7 out of bounds for length 7
t/02-rakudo/19-mkdir.t ...................... ok
t/02-rakudo/17-cli.t ........................ Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/3 subtests
t/02-rakudo/08-repeat.t ..................... ok
t/02-rakudo/18-pseudostash.t ................ Dubious, test returned 3 (wstat 768, 0x300)
Failed 3/3 subtests
t/02-rakudo/dd.t ............................ Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/2 subtests
t/02-rakudo/dump.t .......................... ok
t/02-rakudo/v6.d-tests/01-deprecations.t .... ok
t/02-rakudo/99-misc.t ....................... ok
t/02-rakudo/14-revisions.t .................. Dubious, test returned 3 (wstat 768, 0x300)
Failed 3/3 subtests
t/03-jvm/01-interop.t ....................... ok
t/04-nativecall/01-argless.t ................ ok
t/04-nativecall/02-simple-args.t ............ ok
t/04-nativecall/00-misc.t ................... skipped: NullPointerException in sub NCstrlen
t/04-nativecall/04-pointers.t ............... ok
t/04-nativecall/03-simple-returns.t ......... ok
t/04-nativecall/05-arrays.t ................. skipped: NullPointerException in sub ReturnADoubleArray
t/02-rakudo/reproducible-builds.t ........... ok
t/04-nativecall/07-writebarrier.t ........... skipped: NullPointerException in sub _deref
t/04-nativecall/08-callbacks.t .............. ok
t/04-nativecall/06-struct.t ................. ok
t/04-nativecall/09-nativecast.t ............. ok
t/04-nativecall/10-cglobals.t ............... ok
t/04-nativecall/12-sizeof.t ................. ok
t/04-nativecall/13-union.t .................. ok
t/04-nativecall/11-cpp.t .................... ok
t/04-nativecall/13-cpp-mangling.t ........... ok
t/04-nativecall/14-rw-attrs.t ............... ok
t/04-nativecall/16-rt125408.t ............... ok
t/04-nativecall/16-rt125729.t ............... ok
t/04-nativecall/15-rw-args.t ................ Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 12/24 subtests
t/04-nativecall/17-libnames.t ............... ok
t/04-nativecall/19-function-pointers.t ...... ok
t/04-nativecall/18-routine-sig-sanity.t ..... ok
t/04-nativecall/21-callback-other-thread.t .. ok
t/04-nativecall/22-method.t ................. skipped: NullPointerException in sub ReturnAStruct
t/04-nativecall/20-concurrent.t ............. ok
t/04-nativecall/23-incomplete-types.t ....... ok
t/05-messages/01-errors.t ................... Dubious, test returned 3 (wstat 768, 0x300)
Failed 3/51 subtests
t/05-messages/11-overflow.t ................. skipped: These tests do not throw on JVM backend
t/05-messages/moar/01-errors.t .............. skipped: These tests are only for MoarVM backend
t/05-messages/v6.d-tests/01-errors.t ........ ok
t/05-messages/10-warnings.t ................. Dubious, test returned 6 (wstat 1536, 0x600)
Failed 6/9 subtests
t/02-rakudo/15-gh_1202.t .................... Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/2 subtests
t/05-messages/03-errors.t ................... ok
t/06-telemetry/03-thread.t .................. ok
t/06-telemetry/04-threadpool.t .............. ok
t/06-telemetry/02-usage.t ................... ok
t/08-performance/01-iterators.t ............. ok
t/07-pod-to-text/02-input-output.t .......... Dubious, test returned 1 (wstat 256, 0x100)
No subtests run
t/07-pod-to-text/01-whitespace.t ............ Dubious, test returned 1 (wstat 256, 0x100)
No subtests run
t/08-performance/05-processkeys.t ........... ok
t/08-performance/99-misc.t .................. ok
t/06-telemetry/01-basic.t ................... ok
t/08-performance/02-qast-rewrites.t ......... ok
t/10-qast/00-misc.t ......................... ok
t/05-messages/02-errors.t ................... Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/47 subtests
t/02-rakudo/repl.t .......................... Dubious, test returned 41 (wstat 10496, 0x2900)
Failed 41/47 subtests
	(less 3 skipped subtests: 3 okay)

Test Summary Report
-------------------
t/01-sanity/55-use-trace.t                (Wstat: 256 Tests: 3 Failed: 1)
  Failed test:  3
  Non-zero exit status: 1
t/02-rakudo/10-nqp-ops.t                  (Wstat: 256 Tests: 2 Failed: 1)
  Failed test:  2
  Non-zero exit status: 1
t/02-rakudo/08-slangs.t                   (Wstat: 256 Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
t/02-rakudo/11-deprecated.t               (Wstat: 256 Tests: 2 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
t/02-rakudo/17-cli.t                      (Wstat: 256 Tests: 3 Failed: 1)
  Failed test:  3
  Non-zero exit status: 1
t/02-rakudo/18-pseudostash.t              (Wstat: 768 Tests: 3 Failed: 3)
  Failed tests:  1-3
  Non-zero exit status: 3
t/02-rakudo/dd.t                          (Wstat: 512 Tests: 2 Failed: 2)
  Failed tests:  1-2
  Non-zero exit status: 2
t/02-rakudo/14-revisions.t                (Wstat: 768 Tests: 3 Failed: 3)
  Failed tests:  1-3
  Non-zero exit status: 3
t/04-nativecall/15-rw-args.t              (Wstat: 65280 Tests: 12 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 24 tests but ran 12.
t/05-messages/01-errors.t                 (Wstat: 768 Tests: 51 Failed: 3)
  Failed tests:  10-12
  Non-zero exit status: 3
t/05-messages/10-warnings.t               (Wstat: 1536 Tests: 9 Failed: 6)
  Failed tests:  2-4, 6-8
  Non-zero exit status: 6
t/02-rakudo/15-gh_1202.t                  (Wstat: 512 Tests: 2 Failed: 2)
  Failed tests:  1-2
  Non-zero exit status: 2
t/07-pod-to-text/02-input-output.t        (Wstat: 256 Tests: 0 Failed: 0)
  Non-zero exit status: 1
  Parse errors: No plan found in TAP output
t/07-pod-to-text/01-whitespace.t          (Wstat: 256 Tests: 0 Failed: 0)
  Non-zero exit status: 1
  Parse errors: No plan found in TAP output
t/05-messages/02-errors.t                 (Wstat: 512 Tests: 47 Failed: 2)
  Failed tests:  33, 43
  Non-zero exit status: 2
t/02-rakudo/repl.t                        (Wstat: 10496 Tests: 47 Failed: 41)
  Failed tests:  1-6, 8-28, 33-34, 36-47
  Non-zero exit status: 41
Files=104, Tests=1407, 817 wallclock secs ( 1.08 usr  0.36 sys + 11316.35 cusr 574.89 csys = 11892.68 CPU)
Result: FAIL
make: *** [j-coretest5] Error 1
```
```
% ./rakudo
OpenJDK 64-Bit Server VM warning: Options -Xverify:none and -noverify were deprecated in JDK 13 and will likely be removed in a future release.
Welcome to Rakudo™ v2023.09.
Implementing the Raku® Programming Language v6.d.
Built on JVM.

You may want to `zef install Readline`, `zef install Linenoise`, or `zef install Terminal::LineEditor` or use rlwrap for a line editor

To exit type 'exit' or '^D'
[0] > print "Hello World\n"
Hello World
[0] >
```
